### ISSUE TITLE
Exception when parsing response from contentful

### ISSUE THREAD

**Issue by tafatt** [contentful/contentful.net#226]
Hi,

I get this exception when you SDK is parsing a json response from the delivery API:

`Can not add Newtonsoft.Json.Linq.JValue to Newtonsoft.Json.Linq.JObject.`

I have by manually selecting specific fields identified that it is a field that links to another entity that is the problem. Do you have any ideas what could cause this?

**Comment by Roblinde** [comment]
Hi @tafatt 

What does your model look like that you're deserializing into?

**Comment by Roblinde** [comment]
No they don't have to be, but I'm curious which field is it complaining about? 

There's a `SystemProperties` class you can use for Sys if you want to.

**Comment by tafatt** [comment]
It's a field called Offer. But I don't want to paste the whole data model here.

**Comment by Roblinde** [comment]
If you look at the raw json response from Contentful is there anything unexpected there for the Offer field? Not sure what would cause this.

Could you paste the entire stack trace?

**Comment by tafatt** [comment]
```
fail: Api.Controllers.ArticleRecommendationsController[0]
      Get from contentful failed
      System.ArgumentException: Can not add Newtonsoft.Json.Linq.JValue to Newtonsoft.Json.Linq.JObject.
         at Newtonsoft.Json.Linq.JObject.ValidateToken(JToken o, JToken existing)
         at Newtonsoft.Json.Linq.JContainer.InsertItem(Int32 index, JToken item, Boolean skipParentCheck)
         at Newtonsoft.Json.Linq.JObject.InsertItem(Int32 index, JToken item, Boolean skipParentCheck)
         at Newtonsoft.Json.Linq.JContainer.AddInternal(Int32 index, Object content, Boolean skipParentCheck)
         at Newtonsoft.Json.Linq.JContainer.Add(Object content)
         at Contentful.Core.ContentfulClient.GetEntries[T](String queryString, CancellationToken cancellationToken)
         at Contentful.Core.ContentfulClient.GetEntries[T](QueryBuilder`1 queryBuilder, CancellationToken cancellationToken)
         at Services.ArticleService.Services.IArticleService.GetAll() in /Users/markuslagerstrom/dev/PreglifeArticleRecommendations/Services/ArticleService.cs:line 42
         at Api.Controllers.ArticleRecommendationsController.GetList() in /Users/markuslagerstrom/dev/PreglifeArticleRecommendations/Api/Controllers/ArticleRecommendationsController.cs:line 52
Api.Controllers.ArticleRecommendationsController: Error: Get from contentful failed

System.ArgumentException: Can not add Newtonsoft.Json.Linq.JValue to Newtonsoft.Json.Linq.JObject.
   at Newtonsoft.Json.Linq.JObject.ValidateToken(JToken o, JToken existing)
   at Newtonsoft.Json.Linq.JContainer.InsertItem(Int32 index, JToken item, Boolean skipParentCheck)
   at Newtonsoft.Json.Linq.JObject.InsertItem(Int32 index, JToken item, Boolean skipParentCheck)
   at Newtonsoft.Json.Linq.JContainer.AddInternal(Int32 index, Object content, Boolean skipParentCheck)
   at Newtonsoft.Json.Linq.JContainer.Add(Object content)
   at Contentful.Core.ContentfulClient.GetEntries[T](String queryString, CancellationToken cancellationToken)
   at Contentful.Core.ContentfulClient.GetEntries[T](QueryBuilder`1 queryBuilder, CancellationToken cancellationToken)
   at Services.ArticleService.Services.IArticleService.GetAll() in /Users/markuslagerstrom/dev/PreglifeArticleRecommendations/Services/ArticleService.cs:line 42
   at Api.Controllers.ArticleRecommendationsController.GetList() in /Users/markuslagerstrom/dev/PreglifeArticleRecommendations/Api/Controllers/ArticleRecommendationsController.cs:line 52

```

**Comment by Roblinde** [comment]
hmm only thing I can think of is that the reference field Offer is of an incorrect data type in the model. In your ArticlesDto, when you add the Offer what datatype do you use for that? It should be a class, not a primitive value.

**Comment by tafatt** [comment]
I dont have Offer in ArticlesDto at all. So it's not trying to map it to the class. Maybe we should get in contact some other way and I could give you some responses in a less public channel.

**Comment by Roblinde** [comment]
Sure, mail me at robert@contentful.com and we'll try to get to the bottom of this.

**Comment by Roblinde** [comment]
This should now be fixed in version 6.0.6
